src/utils/types.ts

```diff
@@ -50,18 +50,18 @@ export const ResponsePaymentError = (
     detail,
     detail_v2: detailV2,
     status: pipe(
-      WithinRangeInteger(100, 599).decode(HttpStatusCodeEnum.HTTP_STATUS_500),
+      WithinRangeInteger(100, 599).decode(HttpStatusCodeEnum.HTTP_STATUS_424),
```

Comment:
I suggest to just use a cast when using know constants instead of decoding. You can give the type an alias for ergonomicity (ie. `type HttpCode = WithinRangeInteger<100, 599>`) then
```suggestion
      HttpStatusCodeEnum.HTTP_STATUS_424 as HttpCode,
```